lib/test-file.js

```diff
@@ -0,0 +1,162 @@
+// Based on
+// https://github.com/smikes/test262-parser/blob/04902c58d2e2d1b452b34754924b8e75f2e5f3ac/lib/parser.js
+// Copyright (C) 2014, Microsoft Corporation. All rights reserved.
+// This code is governed by the BSD License found in the LICENSE-MICROSOFT.txt
+'use strict';
+
+const yaml = require('js-yaml');
+const yamlStart = '/*---';
+const yamlEnd = '---*/';
+
+/**
+ * Extract copyright message
+ *
+ * @method extractCopyright
+ * @param {Test262File} file - file object
+ * @return {string} the copyright string extracted from contents
+ * @private
+ */
+function extractCopyright(test262File) {
+  let lines = test262File.contents.split(/\r|\n/);
```

Comment:
I think only if the copyright is on the same line, immediately following the real <PS>, because this works as expected: 

```js
`#!foo 
// Copyright 2001 A Space Odyssey
// There is a real <PS> immediately following '#!foo'`.split(/\r|\n/);
```

Result: 

```
[
  "#!foo ",
  "// Copyright 2001 A Space Odyssey",
  "// There is a real <PS> immediately following '#!foo'",
]
```

However, this would be a problem: 

```js
`#!foo // Copyright 2001 A Space Odyssey
// There is a real <PS> immediately following '#!foo'`.split(/\r|\n/);
```

Result: 

```
[
"#!foo // Copyright 2001 A Space Odyssey",
"// There is a real <PS> immediately following '#!foo'"
]
```

...However... There is presently no restriction in INTERPRETING.md against writing `#!foo // Copyright 2001 A Space Odyssey`. It will also pass lint checks.